src/Avalonia.Base/Utilities/Ref.cs

```diff
@@ -129,12 +146,13 @@ public IRef<T> Clone()
             {
                 lock (_lock)
                 {
-                    lock (_lock)
+                    if (_item != null)
                     {
-                        if (_item != null)
-                            return new Ref<TResult>((TResult) (object) _item, _counter);
-                        throw new ObjectDisposedException("Ref<" + typeof(T) + ">");
+                        TResult castItem = null;
+                        Volatile.Write(ref castItem, (TResult)(object)_item);
```

Comment:
You are using Volatile.Write on a local variable, that doesn't make any sense.

Reply:
Wouldn't that still create a memory barrier when needed? If not, then I'll move the volatile write to the Ref constructor.